### DISCUSSION TITLE
与TeX Live 2023的兼容性

### DISCUSSION THREAD

**Discussion by tk166** [tuna/thuthesis#888] [Q&A] [answered]
尝试使用TeX Live 编译 `thuthesis-example.tex`

使用环境的是Windows10 21H2, TexLive2023, Vscode 1.78, LaTeX Workshop Extension 9.11.4

感谢 @zepinglee的指导，概括在该环境下能够正常进行编译的个人方法如下：

1. 删掉两个Makefile相关的文件
```
Makefile
latexmkrc
```

2. 打开一个Terminal运行命令生成 .cls 文件
```
xetex thuthesis.ins
```

3. 在`appendix.tex`里面注释掉`printbibliography`这一行，改用`bibliography`命令
```latex
% \printbibliography
\bibliography{ref/refs}
```

4. 运行LaTeX Workshop Extension的`Build LaTeX project / Recipe : latexmk (xelatex)` 即可编译成功

> [已解决 1]报错 `Package kvsetkeys: Undefined key 'degree-category'.`
> 需要在`thusetup.tex`里面注释掉了这两行
> ```latex
> % degree-category  = {工学硕士},
> % degree-category* = {Master of Science},
> ```

我是刚看LaTeX教程入坑的小白，知其然不知其所以然。

**Comment by zepinglee** [comment]
> Undefined key 'degree-category'.

这是没有找到最新版的 .cls 文件。建议 `xetex thuthesis.ins` 生成 .cls 文件。

> Undefined control sequence \printbibliography

我尝试修这个问题，但是还没完全解决。

**Reply by tk166** [reply]
对于第二个问题，发现可以使用`\bibliography{ref/refs}`命令代替`\printbibliography`命令来正确编译和显示附录参考文献。请问这种使用方法规范么？

**Reply by zepinglee** [reply, marked as answer]
本来的是 `\bibliography`，打算改成 `\printbibliography`。但是本科生的情况比较复杂，还没有成功。

不过目前本科生应使用 [appendix-survey.tex](https://github.com/tuna/thuthesis/blob/master/data/appendix-survey.tex) 或 [appendix-translation.tex](https://github.com/tuna/thuthesis/blob/master/data/appendix-translation.tex)，不受影响。
